bot/exts/info/doc/_cog.py

```diff
@@ -264,6 +268,25 @@ async def get_symbol_markdown(self, doc_item: DocItem) -> str:
                 return "Unable to parse the requested symbol."
         return markdown
 
+    async def _get_symbols_items(self, symbols: list[str]) -> dict[str, DocItem | None]:
+        """Get DocItems for the given list of symbols, and update the stats for the fetched doc items."""
+        if not self.refresh_event.is_set():
+            log.debug("Waiting for inventories to be refreshed before processing item.")
+            await self.refresh_event.wait()
+
+        # Ensure a refresh can't run in case of a context switch until the with block is exited
```

Comment:
But a context switch cannot happen if there's no `await`/`async for`/`async with`